File: backend/src/main/kotlin/com/example/backend/models/ProduitVendu.kt
package com.example.backend.models

import jakarta.persistence.*

@Entity
@Table(name = "produit_vendu")
class ProduitVendu {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "qtiteVendu")
  var qtiteVendu: Int? = null

  @Column(name = "tva")
  var tva: Double? = 0.0

  @Column(name = "prixUnit")
  var prixUnit: Double? = 0.0

  @Column(name = "montantTTC")
  var montantTTC: Double? = 0.0

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "produit_id")
  var produit: Produit? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "vente_id")
  var vente: com.example.backend.models.Vente? = null
}

File: backend_py/pharmaPython/app/models/produit_vendu.py
from sqlalchemy import Column, Integer, Float, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class ProduitVendu(Base):
  __tablename__ = "produit_vendu"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, index=True)
  qtite_vendu = Column(Integer, nullable=True)
  tva = Column(Float, default=0.0)
  prix_unit = Column(Float, default=0.0)
  montant_ttc = Column(Float, default=0.0)

  produit_id = Column(Integer, ForeignKey("produit.id"))
  produit = relationship("Produit", back_populates="produits_vendus")

  vente_id = Column(Integer, ForeignKey("vente.id"))
  vente = relationship("Vente", back_populates="produits_vendus")
